src/MongoDB.Driver.Core/Core/Operations/ChangeStreamCursor.cs

```diff
@@ -139,6 +182,8 @@ private IEnumerable<TDocument> DeserializeDocuments(IEnumerable<RawBsonDocument>
             var documents = new List<TDocument>();
             RawBsonDocument lastRawDocument = null;
 
+            _postBatchResumeToken = GetPostBatchResumeTokenOrThrow();
```

Comment:
No need for a helper method, just cast directly:

    _postBatchResumeToken = ((ICursorBatchInfo)_cursor).PostBatchResumeToken;

Reply:
Done